lightningd/peer_control.c

```diff
@@ -1390,6 +1409,32 @@ static struct command_result *json_setchannelfee(struct command *cmd,
 		   NULL))
 		return command_param_failed();
 
+	// If the users requested 'all' channels we need to iterate
+	if (json_tok_streq(buffer, idtok, "all")) {
+		count = 0;
+		list_for_each(&cmd->ld->peers, peer, list) {
+			list_for_each(&peer->channels, channel, list) {
+				channel = peer_active_channel(peer);
+				if (!channel)
+					continue;
+				if (channel->state != CHANNELD_NORMAL &&
+					channel->state != CHANNELD_AWAITING_LOCKIN)
+					continue;
+				count++;
+				set_channel_fees(cmd, channel, *base, *ppm);
+			}
+		}
+		/* write response */
+		response = json_stream_success(cmd);
+		json_object_start(response, NULL);
+		json_add_num(response, "base", *base);
+		json_add_num(response, "ppm", *ppm);
+		json_add_string(response, "id", "all");
+		json_add_num(response, "count", count);
+		json_object_end(response);
+		return command_success(cmd, response);
```

Comment:
Should this return an array of results, one for each peer we changed? That implies the normal single-peer version should return an array too (we don't change form of output based on input parameters).

We can do this because API are not stable until release...

Reply:
Thanks, will do

Reply:
addressed by 287a0ed